.rubocop.yml

```diff
@@ -1,63 +1,72 @@
+
+# List of cops can be found here:
+# https://github.com/bbatsov/rubocop/blob/master/config/default.yml
+# https://github.com/bbatsov/rubocop/blob/master/config/disabled.yml
+# https://github.com/bbatsov/rubocop/blob/master/config/enabled.yml
+
 inherit_from: .rubocop_todo.yml
 
 AllCops:
   DisplayCopNames: true
-  Include:
-    - '**/Rakefile'
-    - '**/config.ru'
+  DisplayStyleGuide: true
   Exclude:
     - 'db/**/*'
-    - 'script/**/*'
-    - 'spec/internal/**/*'
-    - 'spec/test_app_templates/**/*'
-    - 'vendor/**/*'
-
-Lint/ImplicitStringConcatenation:
-  Exclude:
-    - 'lib/generators/sufia/**/*'
+    - 'tmp/**/*'
+  ExtraDetails: true
+  TargetRubyVersion: 2.2
 
-Metrics/LineLength:
-  Max: 400
+# Alot of these metrics are quite restrictive. We may want to disable a few of these.
+# For now let's keep them and increase them by an excessive amount
 
+# A calculated magnitude based on number of assignments,
+# branches, and conditions.
 Metrics/AbcSize:
-  Max: 30
-  Exclude:
-    - 'app/controllers/concerns/sufia/admin/depositor_stats.rb'
-    - 'app/controllers/concerns/sufia/my_controller_behavior.rb'
-    - 'app/services/sufia/user_stat_importer.rb'
-    - 'lib/sufia/arkivo/metadata_munger.rb'
+  Max: 45 # default 15
 
+Metrics/ClassLength:
+  Max: 500 # default 100
+
+# A complexity metric that is strongly correlated to the number
+# of test cases needed to validate a method.
+Metrics/CyclomaticComplexity:
+  Max: 17 # default 6
+
+Metrics/LineLength:
+  Enabled: true
+  Max: 120 # default is 80
```

Comment:
Yeah we can go to 80 if everyone wants to. Curious to hear what @mbarnett or others think about this. Don't want to bike shed this though. You can still use your 80 line length, you are not being forced to use 120 or anything like that. The main reasons why I chose 120 is its "good enough" without being to restrictive and also since rails/rails and sufia are not following a proper linelength cop, majority of the generated code is well over this limit. There is currently 130+ errors for having LineLength over 80 which would probably take a couple hours to fix (only about 27+ errors for LineLength at 100).

Yeah totally agree. I looked mainly at sufia and curation_concerns repos and sufia is using a LineLength of 400 (https://github.com/projecthydra/sufia/blob/master/.rubocop.yml#L20 which is silly) and curation_concerns turned to LineLength off (https://github.com/projecthydra/curation_concerns/blob/master/.rubocop.yml#L21  which isn't ideal either). So something sane is better then nothing.